cpp/src/arrow/util/utf8.cc

```diff
@@ -164,5 +176,21 @@ ARROW_EXPORT Result<std::string> WideStringToUTF8(const std::wstring& source) {
   }
 }
 
+ARROW_EXPORT Result<std::string> UTF16StringToUTF8(const std::u16string& source) {
```

Comment:
For the record, `ARROW_EXPORT` is only useful on declarations, not definitions.

Reply:
Oh, sorry. I missed this.

Reply:
Oh that's my bad. I copied the line from the header file but forgot to delete `ARROW_EXPORT`. I'll submit a followup pull request.